server/src/service_manager.rs

```diff
@@ -0,0 +1,26 @@
+// Secret Service Manager
+
+use std::{
+    collections::HashMap,
+    sync::{Arc, Mutex},
+};
+
+use zbus::zvariant::OwnedObjectPath;
+
+use crate::session::Session;
+
+#[derive(Debug, Default)]
+pub struct ServiceManager {
+    // store Secret Sessions mapped to ObjectPaths
+    sessions: Mutex<HashMap<OwnedObjectPath, Arc<Session>>>,
```

Comment:
The service manager itself is already stored behind a Mutex, you don't need an extra one here.

Reply:
done, thanks!